### PR TITLE
Fix problems.

### PR DESCRIPTION
**Pull request by kenmazaika**
There are a few typos thoughout the application.  This will point out the specific changes you need to make in your project.

------

As a side note, this is a GitHub pull request.  This is how I can show you code that I think you should add to your project.  It allows you to look at the code, you'll see the green code is code that's added and the red is code that's deleted.  The "Files changed" tab will show you the code.

"Commits" will show you a listing of the individual commits I'm trying to send you.  And this tab, the conversation tab you can use to communicate about the change.

If you want to suck in all this code into your project you can by pressing the green merge button.  After you press the merge button, you'll have to update your web dev environment to have the latest updates from GitHub and to do that you can run this command:

```
git pull origin master
```

Hope this helps.

### CODE REVIEW

File: app/controllers/instructor/lessons_controller.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ def current_section
     @current_section ||= Section.find(params[:section_id])
   end
 
-  def lesson_parms
+  def lesson_params
```
Comment (kenmazaika) [inline app/controllers/instructor/lessons_controller.rb]: Typo here, it's `lesson_params` not `lesson_parms`.

File: app/views/courses/index.html.erb [hunk 2]
```diff
@@ -10,7 +10,7 @@
   </h2>
   <br />
   <div class="col-12 text-center">
-    <%= image_tag @course.image, class: 'img-fluid' %>
+    <%= image_tag course.image, class: 'img-fluid' %>
```
Comment (kenmazaika) [inline app/views/courses/index.html.erb]: If you look at how you reference the course inside this view file, it is referenced throughout the index as `course` not `@course`.

File: app/views/instructor/courses/show.html.erb [hunk 3]
```diff
@@ -17,7 +17,7 @@
     <% current_course.sections.each do |section| %>
     <li>
       <%= section.title %>
-      <%= link_to 'New Lesson', new_instructor_course_section_path(section), class: 'float-right btn btn-primary btn-sm' %>
+      <%= link_to 'New Lesson', new_instructor_section_lesson_path(section), class: 'float-right btn btn-primary btn-sm' %>
```
Comment (kenmazaika) [inline app/views/instructor/courses/show.html.erb]: The `new lesson` passes along the `section_id` through the parameters, not the `course_id`.

File: app/views/instructor/lessons/new.html.erb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <br />
 <div class="booyah-box col-10 offset-1">
   <h1>Add a new lesson</h1>
-  <%= simple_form_for @lesson url: instructor_section_lessons_path(current_section) do |f| %>
+  <%= simple_form_for @lesson, url: instructor_section_lessons_path(current_section) do |f| %>
```
Comment (kenmazaika) [inline app/views/instructor/lessons/new.html.erb]: Missing comma.

File: app/views/lessons/show.html.erb [hunk 5]
```diff
@@ -0,0 +1 @@
+Lesson Page
```
Comment (kenmazaika) [inline app/views/lessons/show.html.erb]: This file should exist if you want to visit a lesson detail page.  future lessons should flesh out the content.

File: config/routes.rb [hunk 6]
```diff
@@ -10,7 +10,7 @@
       end
       resources :courses, only: [:new, :create, :show] do
         resources :sections, only: [:new, :create]
-          end
-        end
```
Comment (kenmazaika) [inline config/routes.rb]: Optional, but re-indenting the file makes it easier to read.

**Comment by andrejensen302** [comment]
Got it.

Thank you, Ken!

**Pull request merged**